lib/kramdown_rpf/rpf.rb

```diff
@@ -110,6 +112,31 @@ def self.convert_print_only_to_html(content)
         ::Kramdown::Document.new("<div class=\"u-print-only\">\n#{content}</div>", KRAMDOWN_OPTIONS).to_html
       end
 
+      def self.convert_quiz_to_html(quiz)
+        quiz =~ YAML_FRONT_MATTER_REGEXP
+        details = YAML.safe_load(Regexp.last_match(1))
+        question = details['question']
+        choices = YAML.safe_load(Regexp.last_match(2))
```

Comment:
I'd be happier if the regex match here assigned a variable:

```ruby
matches = YAML_FRONT_MATTER_REGEXP.match(quiz)
details = YAML.safe_load(matches[1])
choices  = YAML.safe_load(matches[2])
```

Bonus points if you can do named captures in the regex :)

```ruby
YAML_FRONT_MATTER_REGEXP = /\n\s*---\s*\n(?<details>.*?)---(?<choices>.*)/m
...
matches = YAML_FRONT_MATTER_REGEXP.match(quiz)
details = YAML.safe_load(matches['details'])
choices  = YAML.safe_load(matches['choices'])
```

It just feels a little more deliberate.

Reply:
re: second point, the same regex is already used elsewhere (with different variable names for the match groups) - you could argue for making it consistent throughout but I'm always loathe to fiddle with existing working regex stuff, and then you are making the context-relevant naming into generic stuff